Developer-Guide--Set-Up-With-Docker.md

```diff
@@ -203,14 +203,96 @@ If you plan on doing work that involves sending/recieving emails from MarkUs, yo
+
+Since this is not a wide-spread issue, it's more reasonable to have the setup living entirely on your local (i.e. ignored by git) than committing it to the repo.
+
+1. Start by creating a `docker-compose.override.yml` file under Markus root. Notice that the filename is already listed in `.gitignore`.
+2. The general idea is simple - configure `TMPDIR`, then pass this configuration in. Now we need to find a potential `TMPDIR` candidate inside the container. Reading <https://github.com/ruby/ruby/blob/ruby_3_0/lib/tmpdir.rb> gives us an idea of what ruby expects (at the time of writing, Markus was in ruby 3.0, but this file shouldn't expect major changes in the future versions. If it starts using other env var(s), update this documentation to reflect the new env var(s)).
+3. You can find a directory in the rails container with the correct permissions (1777) & that is unused by Markus, or create your own. In my case `/var/tmp` fit the profile.
+4. Write this env var into the `docker-compose.override.yml` file you created. For example:
+
+    ```YAML
+    version: '3.7'
+
+    services:
+    rails:
+        environment:
+        - TMPDIR=/var/tmp
+    ```
+
+5. Save your changes. After `docker compose build app`, make sure you run `docker compose -f docker-compose.override.yml docker-compose.yml up rails` instead of just `docker compose up rails`. This will apply the `TMPDIR` we created, which would resolve the issue.
+
+### Q3
+
+When the `rails` container is started, postgres' database migrations will be auto applied because of the line `bundle exec rails db:prepare` in `entrypoint-dev-rails.sh`. Sometimes migrations fail - sometimes outright when you first start the container with `docker compose up rails`, other times when you successfully create your `rails` container, then make some data change to markus (i.e. adding a new assignment tag) or shut down and restart the `rails` container - like
+
+```MARKDOWN
+...
+======================================
+2023-09-15 11:47:03 -- create_table(:users, {:id=>:integer})
+2023-09-15 11:47:03 rails aborted!
+2023-09-15 11:47:03 StandardError: An error has occurred, this and all later migrations canceled:
+2023-09-15 11:47:03
+2023-09-15 11:47:03 PG::DuplicateTable: ERROR:  relation "users" already exists
+2023-09-15 11:47:03 /app/db/migrate/20080729160237_create_users.rb:3:in `up'
+2023-09-15 11:47:03
+2023-09-15 11:47:03 Caused by:
+2023-09-15 11:47:03 ActiveRecord::StatementInvalid: PG::DuplicateTable: ERROR:  relation "users" already exists
+...
+```
+
+or other errors typically after the migration `2023-09-15 11:18:50 Assign Marks for Assignments 0-2` is run. Both of these  also occur after following the setup guide step by step.
```

Comment:
Remove this part and just focus on the migration issue. If you encounter the seed error for assigning marks again, please instead make an issue on the MarkUs repo---we should definitely just fix that in the seed files.